Bradley Combat Plugin

bradley-t-t opened the pull request:
… updates I make later on. Some UTIL was updated as well.

chsami committed:
Merge pull request #714 from chsami/development

1.7.5

chsami committed:
Merge pull request #715 from chsami/development

1.7.5.1

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/combat/Rs2Combat.java:
```diff
@@ -37,6 +40,55 @@ public static boolean setAttackStyle(WidgetInfo style) {
+     * @param specialAttackEnergyRequired needed spec energy (1000 = 100%)
+     * @return true if successful or condition is met, false otherwise
+     */
+    public static boolean enableSpecState(int specialAttackEnergyRequired) {
```
It looks as Rs2Combat.setSpecState already does the same action?

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/combat/Rs2Combat.java:
```diff
@@ -37,6 +40,55 @@ public static boolean setAttackStyle(WidgetInfo style) {
+     * @param doubleSpec                  if true, clicks spec twice quickly
+     * @return false if cannot set or not enough energy, otherwise true (once set)
+     */
+    public static boolean enableSpecState(int specialAttackEnergyRequired, boolean doubleSpec) {
```
I think this is logic that can just be handled inside the script, not really needed as a utility method

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/magic/Rs2Magic.java:
```diff
@@ -53,10 +53,36 @@ public static boolean oneTimeSpellBookCheck() {
+     * Attempts to cast Vengeance if the cooldown is over and spell is available.
+     * Integrates with TimersAndBuffs by checking Varbits.VENGEANCE_COOLDOWN first.
+     */
+    public static void castVengeance() {
```
Could you just leverage Rs2Magic.quickCast(MagicAction.VENGEANCE) ?

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/walker/Rs2Walker.java:
```diff
@@ -91,6 +92,23 @@ public static boolean walkTo(WorldPoint target, int distance) {
         return walkWithState(target, distance) == WalkerState.ARRIVED;
     }
 
+    /**
```
This method already exists under Rs2Player.walkUnder()

gmason0 reviewed (changes requested):
1. Can you please review the API for already existing methods, as it seems that some of these additions already exist/unnecessary.

2. Can we please remove all the formatting changes? It makes such a large pull request difficult to review.

3. Can we also merge this into development? We will not accept the PR if its merging to main.

bradley-t-t reviewed (commented):
I believe I did everything you guys requested. Sorry, new to GitHub.

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/bradleycombat/BradleyCombatScript.java:
```diff
@@ -0,0 +1,31 @@
+
+import java.util.concurrent.TimeUnit;
+
+public class BradleyCombatScript extends Script {
```
this script does nothing?

bradley-t-t replied:
This script has planned functionality for eating, praying, potions, etc. Will be added to next update most likely.

bradley-t-t committed:
Bradley combat plugin commit.

Apologies for issues with last commit. Still learning.

bradley-t-t committed:
Bradley combat plugin commit.

Apologies for issues with last commit. Still learning.

bradley-t-t committed:
Merge branch 'chsami:main' into bradley

bradley-t-t committed:
Merge remote-tracking branch 'origin/bradley' into bradley

bradley-t-t committed:
Bradley combat plugin commit.

Made changes based on comments. Integrated the util for targeting into the plugin directly.
